Mathlib/Order/Heyting/Basic.lean

```diff
@@ -1186,6 +1165,20 @@ def LinearOrder.toBiheytingAlgebra [LinearOrder α] [BoundedOrder α] : Biheytin
     top_sdiff := fun a => if_congr top_le_iff rfl rfl }
 #align linear_order.to_biheyting_algebra LinearOrder.toBiheytingAlgebra
 
+instance OrderDual.instBiheytingAlgebra [BiheytingAlgebra α] : BiheytingAlgebra αᵒᵈ where
+  __ := instHeytingAlgebra
+  __ := instCoheytingAlgebra
+
+instance Prod.instBiheytingAlgebra [BiheytingAlgebra α] [BiheytingAlgebra β] :
+    BiheytingAlgebra (α × β) where
+  __ := instHeytingAlgebra
+  __ := instCoheytingAlgebra
+
+instance Pi.instBiheytingAlgebra {α : ι → Type*} [∀ i, BiheytingAlgebra (α i)] :
+    BiheytingAlgebra (∀ i, α i) where
+  __ := instHeytingAlgebra
+  __ := instCoheytingAlgebra
+
```

Comment:
Can you either split this, or mention it in the PR description?

Reply:
#12699 does the split.


As far as I can tell all the other changes in this file are just stylistic, and so probably don't belong in this PR at all?

Reply:
The style changes are my attempt to make sure everything is set up correctly.

I've split them off to #12709.